tests/sentry/models/test_outbox.py

```diff
@@ -31,8 +33,20 @@
 from sentry.testutils.factories import Factories
 from sentry.testutils.outbox import outbox_runner
 from sentry.testutils.region import override_regions
-from sentry.testutils.silo import control_silo_test, exempt_from_silo_limits, region_silo_test
-from sentry.types.region import Region, RegionCategory
+from sentry.testutils.silo import assume_test_silo_mode, control_silo_test, region_silo_test
+from sentry.types.region import Region, RegionCategory, get_local_region
+
+
+def wrap_with_connection_closure(c: Callable[..., Any]) -> Callable[..., Any]:
+    def wrapper(*args: Any, **kwds: Any) -> Any:
+        try:
+            return c(*args, **kwds)
+        finally:
+            for connection in connections.all():
+                connection.close()
+
+    functools.update_wrapper(wrapper, c)
+    return wrapper
```

Comment:
@corps helped me add this to ensure we don't hit a funky cleanup race with the multi-threaded tests.